libvast/test/system/pivoter.cpp

```diff
@@ -0,0 +1,143 @@
+
+template <class Reader>
+std::vector<table_slice_ptr> inhale(const char* data) {
+  auto input = std::make_unique<std::istringstream>(data);
+  Reader reader{vast::defaults::system::table_slice_type, caf::settings{},
+                std::move(input)};
+  std::vector<table_slice_ptr> slices;
+  auto add_slice
+    = [&](table_slice_ptr ptr) { slices.emplace_back(std::move(ptr)); };
+  auto [err, produced] = reader.read(std::numeric_limits<size_t>::max(),
+                                     vast::defaults::system::table_slice_size,
+                                     add_slice);
+  if (err != caf::none && err != vast::ec::end_of_input) {
+    auto str = to_string(err);
+    FAIL("reader returned an error: " << str);
+  }
+  return slices;
+}
+
+struct mock_node_state {
+  std::vector<invocation> invocs;
+  static inline constexpr const char* name = "mock-node";
+};
+
+using mock_node_actor = caf::stateful_actor<mock_node_state>;
+
+caf::behavior mock_node(mock_node_actor* self) {
+  return {[=](invocation invocation) {
+    self->state.invocs.push_back(std::move(invocation));
+  }};
+}
+
+struct fixture : fixtures::deterministic_actor_system {
+  fixture() {
+    MESSAGE("spawn mock node");
+    node = sys.spawn(mock_node);
+    run();
+  }
+
+  ~fixture() {
+    self->send_exit(aut, caf::exit_reason::user_shutdown);
+  }
+
+  void spawn_aut(expression expr, std::string target_type) {
+    aut = sys.spawn(vast::system::pivoter, node, std::move(target_type),
+                    std::move(expr));
+    run();
+  }
+
+  const std::vector<table_slice_ptr> slices
+    = inhale<vast::format::zeek::reader>(zeek_conn_m57_head);
+
+  caf::actor node;
+  caf::actor aut;
+};
+
+} // namespace
+
+FIXTURE_SCOPE(pivoter_tests, fixture)
+
+TEST(count IP point query without candidate check) {
+  MESSAGE("build expression");
+  auto expr = unbox(vast::to<expression>("proto == udp && orig_bytes < 600"));
+  MESSAGE("spawn the pivoter with the target type pcap");
+  spawn_aut(expr, "pcap.packet");
+  MESSAGE("send a table slice");
+  self->send(aut, slices[0]);
+  // The pivoter maps the slice to an expression and passes it on.
+  run();
+  auto& node_state = deref<mock_node_actor>(node).state;
+  REQUIRE_EQUAL(node_state.invocs.size(), 1u);
+  CHECK_EQUAL(
+    node_state.invocs[0].arguments[0],
+    "#type == \"pcap.packet\" && community_id in "
+    "[\"1:aWZfLIquYlCxKGuJ62fQGlgFzAI=\", "
+    "\"1:fLbpXGtS1VgDhqUW+WYaP0v+NuA=\", \"1:BY/pbReW8Oa+xSY2fNZPZUB1Nnk=\", "
+    "\"1:tShwwbRwEMd3S8SvqZxGyvKm+1c=\", \"1:4iHhzk49NeoFdK6VHSCw4ruRbsw=\", "
+    "\"1:+igyiyVnNTFDre/V6pYx89+Lgr8=\", \"1:5NWtNjiw4JPUO4fMM0WobJPFeU8=\", "
+    "\"1:oG55uQUH+XuHYHOFV0c+yOutW8E=\", \"1:FVMx3YawO69eZmiaMJJbrs6447E=\", "
+    "\"1:79fDvfGNCWV1JBYjXCE5Ov1FuMM=\", \"1:aGi0Bt5ApW6HEEO7wfz+PwvniIU=\", "
+    "\"1:JoBDvaK4Tt6BfWSKWPKaJTELr2M=\"]");
```

Comment:
Since the order doesn't matter, I would think that our `set` type is a more natural fit as opposed to a `vector`.